aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1121,11 +1131,11 @@ Tensor amax(const Tensor& self, IntArrayRef dim, bool keepdim) {
 }
 
 Tensor& argmax_out(const Tensor& self, c10::optional<int64_t> dim, bool keepdim, Tensor& result) {
-  TORCH_CHECK(self.numel() > 0, "cannot perform reduction function argmax on a "
-      "tensor with no elements because the operation does not have an identity");
   Tensor in;
   if (dim) {
     auto sizes = self.sizes();
+    zero_numel_check_dims(self, dim.value());
```

Comment:
Something seems a bit questionable here. If I don't specify `dim` for `argmax`, I ought to get an error. I see you have added a new error test for the `dim` case. But what about the `!dim` case? I don't see any error message.

Reply:
Added an error message for !dim case.